tests/checks/default-setup-path.fish

```diff
@@ -0,0 +1,12 @@
+#RUN: %fish -C 'set -g fish %fish' %s
+
+# FIXME: Rationalize behavior when PATH is explicitly unset, should this not behave like PATH=""?
```

Comment:
Answering https://github.com/fish-shell/fish-shell/pull/9990#discussion_r1311176549 by @faho 

> It is very unlikely that that's what people would expect, and a $PATH of "." is basically unusable. If that's explicitly set, sure, whatever - but an unset $PATH can also be caused by misconfiguration or e.g. a thing that just clears the environment.
> 
> $PATH is also something where we should act like other things because it's a system-wide config knob, and other tools also fall back to their default paths - try env -u PATH $shell -c 'echo $PATH', which will print some default for bash, dash and zsh.

IMO: This discussion does not need to be answered now, it is more important to finish the port. I don't think this should block merging this PR, and I don't know why it hasn't been merged yet. If there are any unresolved questions regarding my changes post the initial review I would love to know.

I can open a separate issue and update the comment to link to that. Is that a satisfactionary interim solution?

I think it would make sense to mark the comment I am answering as off-topic, as the change would be introduced by this PR, and not the one that was commented on.

---

Would love to hear if you have a specific example of a situation where you would (un)intentionally unset `PATH`. I am not familiar enough with this situation to determine what is the expected behaviour — hence the comment. `ksh` and `tcsh` appears to not set a default `PATH`, not that those are very relevant.

If this only happens due to misconfigurations, or using fish in non-recommended situations, I think it would be in the spirit of the `fi` in `fish` to still have a default but also print a warning message.

If unsetting `PATH` is done as a way to reset the environment, shouldn't we then nudge the user in the direction of `--no-config` + `PATH=(getconf PATH)`?

Reply:
>IMO: This discussion does not need to be answered now, it is more important to finish the port. I don't think this should block merging this PR

Just to be clear: It doesn't block the PR. It's a note about a misunderstanding I've seen.

I'm not sure why this isn't merged, I think @krobelus, who intended to merge, was just busy.

Or possibly he's just merged it to riir-parser, which I feel woefully unqualified to do anything about. Truth be told I would not have expected it to get that big or be that long-lived.

-------

>If unsetting PATH is done as a way to reset the environment, shouldn't we then nudge the user in the direction of --no-config + PATH=(getconf PATH)?

Not something that people do for a specific shell. You might have a process unset $PATH that's three processes up the chain and doesn't know anything about fish.

Fundamentally the PATH="" case is a consequence of what an empty PATH *component* means, which is the current directory - if you `set PATH "" /usr/bin`, that is exported as `PATH=:/usr/bin`, and any empty component is understood to be the current directory. It's not a "default" in any sense, it's just an alternative way to spell `PATH="."`.

It would be a bad default for an unset PATH with very little use. And since the convention is to use some "proper" default, and we already do, I don't think that a change is helpful.

>If this only happens due to misconfigurations, or using fish in non-recommended situations, I think it would be in the spirit of the fi in fish to still have a default but also print a warning message.

We have, over time, moved away from printing a lot of warnings. We used to print a warning in case someone added a $PATH component that didn't exist. Only it turns out that people like adding $PATH components unconditionally (this was before fish_add_path), and people like using ssh and scp - and scp (the old-school version, don't know about the new sftp-based thing) will launch an actual shell and if that prints anything will abort the transfer.

It would even be possible that some ssh implementation unsets PATH and expects the shell to come up with a good default. We would be throwing quite annoying warnings in that case that our users couldn't do anything about.